### PR TITLE
Add logic to translate regardless of punctuation at the end of the word

### PR DESCRIPTION
**Pull request by jeffinnes**
Fixing issue #63 by using a RegEx to check the last character of each word for any special characters. Once all special characters from the end of the word have been removed and added to a holding variable, translation on the word is attempted. 

After the translation is attempted, the special characters are concatenated back to the end of the word.

This is my first _real_ open source pull request so any feedback would be appreciated!

### CODE REVIEW

File: geeksay.js [hunk 1]
```diff
@@ -155,15 +155,25 @@ function geeksay(text) {
 }
 
 function geeksayWord(text) {
+    const pattern = new RegExp(/[~`!#$%\^&*+=\-\[\]\\';,/{}|\\":<>\?]/);
+    let specialChar = '';
+
+    if (pattern.test(text[text.length - 1])) {
```
Comment (swapagarwal) [inline geeksay.js]: Is this required?

File: geeksay.js [hunk 2]
```diff
@@ -155,15 +155,25 @@ function geeksay(text) {
...
+        while (pattern.test(text[text.length - 1])) {
+            specialChar = text[text.length - 1] + specialChar;
+            text = text.substring(0, text.length -1);
+        }
+    }
+
     if(isNumeric(text)) {
-        return (text >>> 0).toString(2);
+        return (text >>> 0).toString(2) + specialChar;
```
Comment (swapagarwal) [inline geeksay.js]: I like that you have handled numeric case as well.

File: geeksay.js (hunk 1 above)
Comment (jeffinnes) [reply geeksay.js]: Do you mean declaring specialChar as an empty string?

**Comment by jeffinnes** [comment]
Thank you for taking the time to suggest those improvements and give feedback. I'll work on handling leading punctuation and get those changes pushed ASAP!

File: geeksay.js (hunk 1 above)
Comment (swapagarwal) [reply geeksay.js]: I meant the extra `if` check before the `while`.

**Comment by jeffinnes** [comment]
I'm not totally sure I went about this the right way, but I rebased my fork's master with the upstream and the rebased my dev branch off of that to catch everything up.

**Comment by swapagarwal** [comment]
Resolved in #101

**Pull request closed without merging**